src/rgw/rgw_rest_pubsub.cc

```diff
@@ -494,11 +494,11 @@ void RGWPSListTopicsOp::execute(optional_yield y) {
 
   const RGWPubSub ps(driver, get_account_or_tenant(s->owner.id), *s->penv.site);
   if (rgw::all_zonegroups_support(*s->penv.site, rgw::zone_features::notification_v2) &&
-      driver->stat_topics_v1(s->bucket->get_tenant(), null_yield, this) == -ENOENT) {
-    op_ret = ps.get_topics_v1(this, result, y);
-  } else {
+      driver->stat_topics_v1(s->bucket_tenant, null_yield, this) == -ENOENT) {
```

Comment:
* why did you switch the if/else conditions?
* this will fix the issue, but won't work in case of tenented topics (the "bucket_tenant" is probably empty in SNS requests). i think that the right things would be to use the `get_account_or_tenant()` call from line 495 to get the right value

Reply:
1. I switched the conditions based on the logic I observed in `rgw_admin.cc`. In that file, if both conditions are true then `get_topics_v2` is called. From what I understood, these conditions check if the zonegroups support V2 notifications and if there’s no object holding topics in v1. I applied this order here to follow a similar approach.
2. I see your point about tenanted topics. I’ll update the code to use `get_account_or_tenant()` to fetch the correct tenant information for SNS requests.

Reply:
Thank you for the review!
I switched the conditions to fix the issue with V2 support. When both conditions are true:

  - Zonegroups support `V2` notifications.
  - The `V1` topic file is missing.

In this case, we’re fully in `V2`, so the system uses the `get_topics_v2` function to retrieve topics from the `V2` configuration.

If only one condition is met, it can indicate one of the following states:

- **Migration:** The `V1` topic file is exists, but all zonegroups support `V2`, indicating we’re migrating to `V2`. Here, the system should prioritize `get_topics_v1`.
- **V1:** The `V1` topic file exists, and not all zonegroups support `V2`. In this case, we fall back to `get_topics_v1` to retrieve topics from the `V1` configuration.

The updated code now correctly prioritizes `get_topics_v2` when conditions allow and falls back to `get_topics_v1` only when necessary.

I  understand your concern regarding tenanted topics and will update the code to use `get_account_or_tenant()` to fetch the correct tenant information for SNS requests.